[19.07, master] Asterisk 16.x is not build with ASLR PIE

Maintainer: @jslachta 

When PKG_ASLR_PIE is enabled for the build Asterisk 16.x can not be compiled due to this error. Most probably similar to [#9845](https://github.com/openwrt/packages/pull/9845) from packages feed.

Build log is same for OpenWrt 19.07 and OpenWrt master:
```
/-----/build/staging_dir/toolchain-arm_cortex-a9+vfpv3_gcc-7.5.0_musl_eabi/lib/gcc/arm-openwrt-linux-muslgnueabi/7.5.0/../../../../arm-openwrt-linux-muslgnueabi/bin/ld: ast_expr2fz.o: relocation R_ARM_MOVW_ABS_NC against `stdin' can not be used when making a shared object; recompile with -fPIC
ast_expr2fz.o: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
Makefile:182: recipe for target 'check_expr2' failed
make[4]: *** [check_expr2] Error 1
```
By default PKG_ASLR_PIE is in the default settings disabled and it is possible that it can be enabled in the future. 

As workaround till the issue is find and fixed, `PKG_ASLR_PIE:=0` should be applied to the Makefile.

--- a/net/asterisk-16.x/Makefile
+++ b/net/asterisk-16.x/Makefile
@@ -10,7 +10,7 @@ include $(TOPDIR)/rules.mk
 AST_MAJOR_VERSION:=16
 PKG_NAME:=asterisk$(AST_MAJOR_VERSION)
 PKG_VERSION:=$(AST_MAJOR_VERSION).6.1
-PKG_RELEASE:=3
+PKG_RELEASE:=4
 
 PKG_SOURCE:=asterisk-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://downloads.asterisk.org/pub/telephony/asterisk/releases

--- a/net/asterisk-16.x/patches/056-fix-check_expr2-build.patch
+++ b/net/asterisk-16.x/patches/056-fix-check_expr2-build.patch
@@ -1,6 +1,15 @@
 --- a/utils/Makefile
 +++ b/utils/Makefile
-@@ -187,7 +187,6 @@ check_expr2: $(ASTTOPDIR)/main/ast_expr2
+@@ -180,14 +180,13 @@ conf2ael: conf2ael.o ast_expr2f.o ast_ex
+ 
+ check_expr2: $(ASTTOPDIR)/main/ast_expr2f.c $(ASTTOPDIR)/main/ast_expr2.c $(ASTTOPDIR)/main/ast_expr2.h astmm.o
+ 	$(ECHO_PREFIX) echo "   [CC] ast_expr2f.c -> ast_expr2fz.o"
+-	$(CC) -g -c -I$(ASTTOPDIR)/include -DSTANDALONE $(ASTTOPDIR)/main/ast_expr2f.c -o ast_expr2fz.o
++	$(CC) -g -c -I$(ASTTOPDIR)/include $(_ASTCFLAGS) $(ASTTOPDIR)/main/ast_expr2f.c -o ast_expr2fz.o
+ 	$(ECHO_PREFIX) echo "   [CC] ast_expr2.c -> ast_expr2z.o"
+-	$(CC) -g -c -I$(ASTTOPDIR)/include -DSTANDALONE2 $(ASTTOPDIR)/main/ast_expr2.c -o ast_expr2z.o
++	$(CC) -g -c -I$(ASTTOPDIR)/include $(_ASTCFLAGS) -DSTANDALONE2 $(ASTTOPDIR)/main/ast_expr2.c -o ast_expr2z.o
+ 	$(ECHO_PREFIX) echo "   [LD] ast_expr2fz.o ast_expr2z.o  -> check_expr2"
  	$(CC) -g -o check_expr2 ast_expr2fz.o ast_expr2z.o astmm.o -lm $(_ASTLDFLAGS)
  	$(ECHO_PREFIX) echo "   [RM] ast_expr2fz.o ast_expr2z.o"
  	rm ast_expr2z.o ast_expr2fz.o

